src/app/domain/osReceive/services/OsReceiveData.ts

```diff
@@ -38,18 +41,36 @@ const processReceivedFiles = (
   return filesToUpload
 }
 
-const mapFilesToUploadToArray = (
-  filesMap: Map<string, ReceivedFile>
-): OsReceiveFile[] => {
-  return Array.from(filesMap.values()).map(file => ({
-    name: file.fileName,
+const decodeFileName = (fileName: string): string => decodeURI(fileName)
+
+const determineMimeType = (fileName: string, mimeType: string): string =>
+  getMimeType({
+    name: decodeFileName(fileName),
+    mimeType: Platform.OS === 'android' ? mimeType : undefined
+  } as Media)
+
+const createOsReceiveFile = (file: ReceivedFile): OsReceiveFile => {
+  const decodedFileName = decodeFileName(file.fileName)
+  const mimeType = determineMimeType(file.fileName, file.mimeType)
```

Comment:
I remember there was a diff on checking mimeType for Android received files and for iOS received files, did you check that on both platform?

Reply:
> I remember there was a diff on checking mimeType for Android received files and for iOS received files, did you check that on both platform?

yes, on Android I just use `mimeType` directly since it's correct. Only on iOS I try to guess it

Reply:
Ok and so there is no diff on the name part?

Reply:
> Only on iOS I try to guess it

Don't you fixed that here? https://github.com/cozy/cozy-flagship-app/pull/1023/files#diff-e5fbd9f4b85d62986896fb1d8408ad2d8d5a0f2267b5e0b22a9caef7c97255bb

Reply:
> > Only on iOS I try to guess it
> 
> Don't you fixed that here? https://github.com/cozy/cozy-flagship-app/pull/1023/files#diff-e5fbd9f4b85d62986896fb1d8408ad2d8d5a0f2267b5e0b22a9caef7c97255bb

that's true, the Platform checking is then redundant